components/director/internal/domain/api/service.go

```diff
@@ -201,13 +228,24 @@ func (s *service) GetFetchRequest(ctx context.Context, apiDefID string) (*model.
 	return fetchRequest, nil
 }
 
-func (s *service) createFetchRequest(ctx context.Context, tenant string, in model.FetchRequestInput, parentObjectID string) (*string, error) {
+func (s *service) createFetchRequest(ctx context.Context, tenant string, in model.FetchRequestInput, parentObjectID string) (*model.FetchRequest, error) {
 	id := s.uidService.Generate()
 	fr := in.ToFetchRequest(s.timestampGen(), id, tenant, model.APIFetchRequestReference, parentObjectID)
 	err := s.fetchRequestRepo.Create(ctx, fr)
 	if err != nil {
 		return nil, errors.Wrapf(err, "while creating FetchRequest for %s with ID %s", model.APIFetchRequestReference, parentObjectID)
 	}
 
-	return &id, nil
+	return fr, nil
+}
+
+func (s *service) handleFetchRequest(ctx context.Context, api *model.APIDefinition, fr *model.FetchRequest) {
+	var err error
+	api.Spec.Data, err = s.fetchRequestService.FetchAPISpec(fr)
+	if err != nil {
+		fr.Status.Condition = model.FetchRequestStatusConditionFailed
+	} else {
+		fr.Status.Condition = model.FetchRequestStatusConditionSucceeded
```

Comment:
Should the condition be set to `SUCCEEDED` when invalid fetch mode / auth was specified and spec wasn't fetched? I think `INITIAL` condition fits that state better.

Reply:
or even error with proper message